demo/blocks_plug/run.py

```diff
@@ -22,13 +22,12 @@ def change_tab():
 
 with gr.Blocks() as demo:
```

Comment:
I'm not sure if the changes to Tabs are truly backwards compatible.

Changing the demo back to what it was before breaks (shows red `Error`)

![image](https://user-images.githubusercontent.com/41651716/187256296-bfc6ab21-6a7e-4b1f-a6f1-ddc0bb9dec70.png)

```
index.9e34751b.js:6412 TypeError: Cannot read properties of undefined (reading 'type')
    at walk_layout (index.9e34751b.js:6385:57)
    at index.9e34751b.js:6394:45
    at Array.map (<anonymous>)
    at walk_layout (index.9e34751b.js:6394:36)
    at async Promise.all (:7860/index 1)
    at async walk_layout (index.9e34751b.js:6394:4)
    at async Promise.all (:7860/index 0)
    at async walk_layout (index.9e34751b.js:6394:4)
    at async Promise.all (:7860/index 0)
    at async walk_layout (index.9e34751b.js:6394:4)
```

For reference, I just made the following modification to this code block

```python
with gr.Blocks() as demo:
    gr.Markdown("Three demos in one!")
    with gr.Tabs(selected=1) as tabs:
        with gr.Tab("Text Identity", id=0):
            identity_demo.render()
        with gr.Tab("Text Input", id=1):
            input_demo.render()
        with gr.Tab("Text Static", id=2):
            output_demo.render()
    btn = gr.Button("Change tab")
    btn.click(inputs=None, outputs=tabs, fn=change_tab)
```

Reply:
Took me forever to figure out what was going on here! Turns out Interfaces embedded in Blocks broke, finally fixed in by moving the expected_parent logic int the BlockContext.